### PR TITLE
VDS-149: Add segment main parameters blade

### PR DESCRIPTION
**Pull request by ilyawzrd**
Снимок экрана 2020-11-19 в 15 46 45

### CODE REVIEW

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/mainParameters.tpl.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                            </div>
+                        </div>
+                    </div>
+
+                    <va-metaform registered-inputs="blade.metaFields" blade="blade" column-count="1" ng-if="blade.currentEntity"></va-metaform>
+                </fieldset>
+            </form>
+        </div>
+    </div>
+</div>
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/mainParameters.tpl.html]: Add empty string

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/mainParameters.tpl.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                                            <span ng-bind-html="x.name | highlight: $select.search"></span>
+                                        </ui-select-choices>
+                                    </ui-select>
+                                </div>
+                                <div class="list-descr">{{ 'demoSolutionFeaturesModule.blades.customer-segments-parameters.descriptions.store' | translate }}</div>
+                            </div>
+                        </div>
+                    </div>
+
+                    <va-metaform registered-inputs="blade.metaFields" blade="blade" column-count="1" ng-if="blade.currentEntity"></va-metaform>
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/mainParameters.tpl.html]: Why we needed metaform here?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/mainParameters.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+angular.module('virtoCommerce.demoSolutionFeaturesModule')
+    .controller('virtoCommerce.demoSolutionFeaturesModule.customerSegmentsParametersController', ['$scope', 'platformWebApp.bladeNavigationService', 'virtoCommerce.storeModule.stores', function ($scope, bladeNavigationService, stores) {
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/mainParameters.js]: storeService

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js [hunk 4]
```diff
@@ -1,16 +1,47 @@
...
+                blade.activeBladeId = parametersBlade.id;
+                bladeNavigationService.showBlade(parametersBlade, blade);
+            };
+
+            $scope.$watch('blade.currentEntity', (data) => {
+                if (data) {
+                    $scope.totalPropertiesCount = 3;
+                    $scope.filledPropertiesCount = 0;
+
+                    $scope.filledPropertiesCount += blade.currentEntity.startDate ? 1 : 0;
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js:42]: Add IsActive field

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js (hunk 4 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js:42]: @pushnitsa  By default `Is Active` is false and not nullable. When we should treat it as filled and when as isn't ? :)

**Review by asvishnyakov — DISMISSED** [review]
I don't like some things like properties filling counting, but I see the same in Associations and unable to propose better solution. So just accepted, nothing criminal

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js (hunk 4 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js:42]: @ilyawzrd Create segment with all filled properties and IsActive = true, then change it to false (disable segment). Now 3 of 4 properties filled

Comment (pushnitsa) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js:42]: It is come to filled when we opened main parameters blade and click "Ok" button.

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**